frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1055,6 +1058,57 @@ pub mod pallet {
 			<QueuedSolution<T>>::put(solution);
 			Ok(())
 		}
+
+		/// Challenge a solution in the signed phase
+		///
+		/// If the claimed score is correct, the challenger would be slashed.
+		///
+		/// If the claimed score is not correct, the malicious submitter will lose the deposit,
+		/// the solution will be ejected and the challenger will receive some reward
+		/// less than the solution's deposit.
+		// TODO: Add Proper Benchmarks
+		#[pallet::weight(100)]
+		pub fn challenge_solution(origin: OriginFor<T>, index: u32) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			if let Some(submission) = Self::signed_submissions().get_submission(index) {
+				ensure!(
+					T::Currency::can_slash(&who, submission.deposit),
+					<Error<T>>::SignedCannotPayDeposit
+				);
+				match Self::feasibility_check(
+					submission.raw_solution.clone(),
+					ElectionCompute::Signed,
+				) {
+					Ok(solution) => {
+						let _ = T::Currency::slash(&who, submission.deposit);
+						<QueuedSolution<T>>::put(solution);
+						let mut signed_submissions = Self::signed_submissions();
+						let _ = signed_submissions.pop(submission.raw_solution.score);
+
+						Self::deposit_event(Event::Challenged { account: who, outcome: false });
+
+						Ok(())
+					},
+					Err(_error) => {
+						T::Currency::repatriate_reserved(
+							&submission.who,
+							&who,
+							T::ChallengeDepositDiff::get() * submission.deposit,
+							Free,
+						)?;
+						let mut signed_submissions = Self::signed_submissions();
```

Comment:
you should not get this again, reuse the old one that you get before. 

also, this type is not persisted in storage unless you call `put`.